### PR TITLE
Adding netty's HexDump option to Vertx ClientOptions https://github.com/netty/netty/pull/11420

### PR DESCRIPTION
**Pull request by Unev**
Motivation:

Able/Disable the hex dump of messages in Netty's LoggingHandler.

By default, in Vertx, when we enable the logging enable, netty dumps the data in Hex format along with other log level messages.

Dumping of the data in Hex format may not always be needed, so added a new constructor for this here
https://github.com/netty/netty/pull/11420

In this commit, just added the enable HexDump in all the config locations with default value = true, will try to use the new constructor in the next commit.

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: src/main/java/io/vertx/core/net/NetworkOptions.java [hunk 1]
```diff
@@ -68,6 +74,7 @@ public NetworkOptions() {
     reuseAddress = DEFAULT_REUSE_ADDRESS;
     trafficClass = DEFAULT_TRAFFIC_CLASS;
     logActivity = DEFAULT_LOG_ENABLED;
+    hexDumpEnabled = DEFAULT_HEXDUMP_ENABLED
```
Comment (Unev) [inline src/main/java/io/vertx/core/net/NetworkOptions.java:77]: `hexDumpEnabled = DEFAULT_HEXDUMP_ENABLED;`

Comment (Unev) [reply src/main/java/io/vertx/core/net/NetworkOptions.java:77]:
```suggestion
    hexDumpEnabled = DEFAULT_HEXDUMP_ENABLED;
```

**Comment by vietj** [comment]
it is not clear how this relates to the current logging option

**Comment by Unev** [comment]
Current logging Option, if enabled, prints all the log messages along with data, which is being transformed,  in hexadecimal format.

And let's say If I want to have all the logging messages but not the hex dumping of data, then it is not possible currently.

So introducing one more option to have the hexadump enabled or not. Defaults to enable.

**Comment by vietj** [comment]
ok I see more clearly now thanks

**Comment by Unev** [comment]
Hi @vietj , could you please havelook at this again?

This is needed where we want to log the network activity without dumping the message that is being transferred.
thanks,

**Comment by vietj** [comment]
This PR is not complete, it only add an option but this option is never used.

**Comment by Unev** [comment]
Hi @vietj, 
Added use cases for the above optional field.

Thanks

**Comment by Unev** [comment]
Hi @vietj , can you pls have a look at this.

**Comment by Unev** [comment]
Hi @vietj , can you please have look at this.

Thanks,

**Comment by vietj** [comment]
so looking at this, I believe it is better to replace the `hexDumpEnabled` boolean by the actual enum provided in Netty and name it _activityLogFormat_ with the default being `SIMPLE`

**Comment by vietj** [comment]
ping @Unev

**Comment by Unev** [comment]
@vietj, Ack. Will do the required changes.

Thanks.

**Comment by vietj** [comment]
thanks for the contribution

**Comment by Unev** [comment]
Thank you.

**Pull request merged**